api/api.go

```diff
@@ -56,53 +58,56 @@ func CreateDatasetAPI(host, bindAddr, secretKey string, dataStore store.DataStor
 	api.router.HandleFunc("/datasets/{id}", api.getDataset).Methods("GET")
-	api.router.HandleFunc("/datasets/{id}", api.privateAuth.Check(api.putDataset)).Methods("PUT")
 	api.router.HandleFunc("/datasets/{id}/editions", api.getEditions).Methods("GET")
 	api.router.HandleFunc("/datasets/{id}/editions/{edition}", api.getEdition).Methods("GET")
 	api.router.HandleFunc("/datasets/{id}/editions/{edition}/versions", api.getVersions).Methods("GET")
 	api.router.HandleFunc("/datasets/{id}/editions/{edition}/versions/{version}", api.getVersion).Methods("GET")
-	api.router.HandleFunc("/datasets/{id}/editions/{edition}/versions/{version}", api.privateAuth.Check(versionPublishChecker.Check(api.putVersion))).Methods("PUT")
 	api.router.HandleFunc("/datasets/{id}/editions/{edition}/versions/{version}/metadata", api.getMetadata).Methods("GET")
 	api.router.HandleFunc("/datasets/{id}/editions/{edition}/versions/{version}/dimensions", api.getDimensions).Methods("GET")
 	api.router.HandleFunc("/datasets/{id}/editions/{edition}/versions/{version}/dimensions/{dimension}/options", api.getDimensionOptions).Methods("GET")
 
-	instanceAPI := instance.Store{Host: api.host, Storer: api.dataStore.Backend}
-	instancePublishChecker := instance.PublishCheck{Datastore: dataStore.Backend}
-	api.router.HandleFunc("/instances", api.privateAuth.Check(instanceAPI.GetList)).Methods("GET")
-	api.router.HandleFunc("/instances", api.privateAuth.Check(instanceAPI.Add)).Methods("POST")
-	api.router.HandleFunc("/instances/{id}", api.privateAuth.Check(instanceAPI.Get)).Methods("GET")
-	api.router.HandleFunc("/instances/{id}", api.privateAuth.Check(instancePublishChecker.Check(instanceAPI.Update))).Methods("PUT")
-	api.router.HandleFunc("/instances/{id}/dimensions/{dimension}", api.privateAuth.Check(instancePublishChecker.Check(instanceAPI.UpdateDimension))).Methods("PUT")
-	api.router.HandleFunc("/instances/{id}/events", api.privateAuth.Check(instanceAPI.AddEvent)).Methods("POST")
-	api.router.HandleFunc("/instances/{id}/inserted_observations/{inserted_observations}",
-		api.privateAuth.Check(instancePublishChecker.Check(instanceAPI.UpdateObservations))).Methods("PUT")
-	api.router.HandleFunc("/instances/{id}/import_tasks", api.privateAuth.Check(instancePublishChecker.Check(instanceAPI.UpdateImportTask))).Methods("PUT")
-
-	dimension := dimension.Store{Storer: api.dataStore.Backend}
-	api.router.HandleFunc("/instances/{id}/dimensions", dimension.GetNodes).Methods("GET")
-	api.router.HandleFunc("/instances/{id}/dimensions", api.privateAuth.Check(instancePublishChecker.Check(dimension.Add))).Methods("POST")
-	api.router.HandleFunc("/instances/{id}/dimensions/{dimension}/options", dimension.GetUnique).Methods("GET")
-	api.router.HandleFunc("/instances/{id}/dimensions/{dimension}/options/{value}/node_id/{node_id}",
-		api.privateAuth.Check(instancePublishChecker.Check(dimension.AddNodeID))).Methods("PUT")
-
+	if cfg.EnablePrivateEnpoints {
```

Comment:
Can you add log output here to say "Private endpoints enabled..." will make it explicit what "mode" that app is running in from the start up logs.

Reply:
Done